### ISSUE TITLE
src/lex.l: fatal error: poly.tab.h: No such file or directory

### ISSUE THREAD

**Issue by barracuda156** [Zeta611/polycalc#1]
```
src/lex.l:4:10: fatal error: poly.tab.h: No such file or directory
    4 | #include "poly.tab.h"
      |          ^~~~~~~~~~~~
```

There is no such a header in the source.

**Comment by Zeta611** [comment]
Bison should generate it. Did you `make`?

**Comment by barracuda156** [comment]
> Bison should generate it. Did you `make`?

Yes, but I relied on Macports implementation. I will try doing it manually.
Bison is installed and dependency on it was declared in portfile.

**Comment by barracuda156** [comment]
@Zeta611 Indeed, setting `build.cmd` to `make` fixed the build. Only few warnings are there:
```
build/./src/lex.yy.c:1199:16: warning: 'input' defined but not used [-Wunused-function]
 1199 | #else
      |                ^    
build/./src/lex.yy.c:1156:17: warning: 'yyunput' defined but not used [-Wunused-function]
 1156 | 
      |                 ^      
build/./src/poly.tab.c: In function 'yyparse':
build/./src/poly.tab.c:1074:16: warning: implicit declaration of function 'yylex' [-Wimplicit-function-declaration]
 1074 |       yychar = yylex ();
      |                ^~~~~
build/./src/poly.tab.c:1371:7: warning: implicit declaration of function 'yyerror'; did you mean 'yyerrok'? [-Wimplicit-function-declaration]
 1371 |       yyerror (env, verbose, YY_("syntax error"));
      |       ^~~~~~~
      |       yyerrok
```

**Comment by barracuda156** [comment]
@Zeta611 Hmm, but with Clang again same error: https://github.com/macports/macports-ports/actions/runs/3826812968/jobs/6510900973

**Comment by Zeta611** [comment]
I just tested on my Mac, and I can reproduce this issue. As noted in the `README`, I only tested on a Linux machine. I'll take a look at this issue as well.

**Comment by Zeta611** [comment]
I made a dirty fix for this: this manually runs Bison (GNU Bison 3.8.2 installed via `brew`) for `poly.tab.h` and make the compiler ignore `impllicit-function-declaration` error.

There should be a way to fix this by properly configuring dependencies in a Makefile, but I need some help for this.

Does it work now? I tested it on macOS 13.0.1.

**Comment by barracuda156** [comment]
@Zeta611 Thank you, I will try today and update you.

Makefiles is not something I am comfortable with to fine-tune, but I can implement manual running of bison in pre-build stage in Macports.

**Comment by barracuda156** [comment]
So yes, with your fix it builds successfully, both on my 10.6.8 Rosetta locally and on new macOS with Clang on Macports buildbots: https://github.com/macports/macports-ports/pull/17179

**Comment by Zeta611** [comment]
That’s good to hear!
